crates/rattler_installs_packages/src/index/http.rs

```diff
@@ -268,28 +280,44 @@ struct CacheData {
     url: Url,
 }
 
-/// Fill the cache with the
-fn fill_cache<R: Read>(
-    policy: &CachePolicy,
-    url: &Url,
-    mut body: R,
-    handle: FileLock,
-) -> Result<impl Read + Seek, std::io::Error> {
-    let mut cache_writer = handle.begin()?;
-    ciborium::ser::into_writer(
-        &CacheData {
-            policy: policy.clone(),
-            url: url.clone(),
-        },
-        &mut cache_writer,
-    )
-    .map_err(|e| std::io::Error::new(std::io::ErrorKind::InvalidData, e))?;
-    let body_start = cache_writer.stream_position()?;
-    std::io::copy(&mut body, &mut cache_writer)?;
-    drop(body);
-    let body_end = cache_writer.stream_position()?;
-    let cache_entry = cache_writer.commit()?.detach_unlocked();
-    SeekSlice::new(cache_entry, body_start, body_end)
+/// Write cache BOM and return it's current position after writing
+/// BOM is represented by:
+/// [BOM]--[VERSION]--[SIZE_OF_HEADERS_STRUCT]
+fn write_cache_bom<W: Write + Seek>(
+    writer: &mut W,
+    bom_key: &str,
+    version: u8,
+) -> Result<u64, std::io::Error> {
+    writer.write_all(bom_key.as_bytes())?;
+    writer.write_all(&[version])?;
+    writer.stream_position()
+}
```

Comment:
I would not make the `version` and `size` part of the BOM but call that metadata or something.

Reply:
agree- I've renamed it